test/unit/history/invoke.test.js

```diff
@@ -0,0 +1,23 @@
+import History from '../../../src/history'
+import Microcosm from '../../../src/microcosm'
+
+describe('History::invoke', function() {
+
+  it('it does invoke a repo that no longer exists', function() {
+    let parent = new Microcosm()
+    let child = parent.fork()
+
+    parent.on('change', () => child.teardown())
+
+    jest.spyOn(child, 'release')
+
+    parent.addDomain('test', {
+      getInitialState() {
+        return false
+      }
+    })
+
+    expect(child.release).not.toHaveBeenCalled()
+  })
```

Comment:
I'm not super familiar with the lifecycle hooks, but reading this test it isn't obvious to me why the child goes away (child.teardown()?) triggered by adding a domain on the parent? And if so, why would `child.release` not be called? What is `child.release` ?